src/receipt/script_result.rs

```diff
@@ -0,0 +1,34 @@
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde-types-minimal",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum ScriptExecutionResult {
+    Success,
+    Revert,
+    Panic,
+    // Generic failure case since any u64 is valid here
+    GenericFailure(u64),
```

Comment:
I'm not sure its worthy to add a variant only for convert convenience. This will force the user to be aware of this generic failure when it doesn't really add a context by itself.

If the purpose of this is to increase performance, then these descriptive enums should be represented as `u8` with all variants `[0, u8::MAX]` covered. Otherwise, consider using `TryFrom` instead.

Reply:
I originally started with try from but then realized there's no rules in the specs disallowing higher values. So if someone manually serialized this value, it should still be an error right?

Reply:
Hmm but then we are leaking this to the consumers. But as discussed in slack, we can proceed this way and in the future add more details to this script result type.